### PR TITLE
Validation for importing firmware

### PR DESCRIPTION
**Pull request by wellitonSouza**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Fixed firmware import, where an error occurred when the file was different from .HEX

* **What is the current behavior?** (You can also link to an open issue here)
interrupts the system when sending a file other than .HEX
returns error message when not a .HEX file

* **What is the new behavior (if this is a feature change)?**
returns error message when not a .HEX file

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#1165

* **Other information**:

### CODE REVIEW

File: src/js/views/devices/Actions/DeviceHandlerHelper.js [hunk 1]
```diff
@@ -44,11 +44,10 @@ class DeviceHandlerHelper {
 
                     if (filteredAttr[0]) {
                         const attrElement = filteredAttr[0];
-                        if (attrElement) {
```
Comment (mprevide) [inline src/js/views/devices/Actions/DeviceHandlerHelper.js:47]: Are you sure it is necessary to remove this if?

File: src/js/views/devices/Actions/DeviceHandlerHelper.js [hunk 2]
```diff
@@ -44,11 +44,10 @@ class DeviceHandlerHelper {
```
Comment (mprevide) [inline src/js/views/devices/Actions/DeviceHandlerHelper.js:44]: Is it really necessary to change anything in the DeviceHandlerHelper.js file?

File: src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/ImageList.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ const ImageList = ({
                             className="attribute-type"
```
Comment (mprevide) [inline src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/ImageList.js:26]: Is it really necessary to change anything in the ImageList.js file?

File: src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/SidebarFirmImages.js [hunk 4]
```diff
@@ -37,7 +37,13 @@ class SidebarFirmImages extends Component {
     }
 
     onDrop(files, image) {
-        ImageActions.updateImageData(image.id, 'file', files);
+        const fileName = files[0].name.substring(files[0].name.length - 4,files[0].name.length);
```
Comment (mprevide) [inline src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/SidebarFirmImages.js:40]: A space is required after ','. (comma-spacing)

File: src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/index.js [hunk 5]
```diff
@@ -65,6 +65,7 @@ class SidebarFirmware extends Component {
                         toaster.success('Image updated.');
                     }
```
Comment (mprevide) [inline src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/index.js:67]: Is it really necessary to change anything in the Sidebar/SidebarFirmware/index.js file?

File: src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/SidebarFirmImages.js (hunk 4 above)
Comment (mprevide) [inline src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/SidebarFirmImages.js:40]: Check before: 
if `files` is array, if has property name and length isn't not nul

```js
if(files && Array.isArray(files)&& files.length>0){
  if (files[0] && files[0].hasOwnProperty('name') && files[0].name && files[0].name.length > 4 ) {


```

File: src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/SidebarFirmImages.js [hunk 6]
```diff
@@ -37,7 +37,13 @@ class SidebarFirmImages extends Component {
     }
```
Comment (mprevide) [inline src/js/views/templates/TemplateList/Sidebar/SidebarFirmware/SidebarFirmImages.js:37]: I think you need just commit this js file.

File: src/js/views/devices/Actions/DeviceHandlerHelper.js (hunk 1 above)
Comment (wellitonSouza) [reply src/js/views/devices/Actions/DeviceHandlerHelper.js:47]: yes because it is the comparison of the same value of line 47

Comment (mprevide) [reply src/js/views/devices/Actions/DeviceHandlerHelper.js:47]: But you don't need to fix this to resolve your issue, the right thing is to fix only your issue, for other corrections you would have to open another pull request.
But either way line 47 is empty

**Pull request closed without merging**